src/sentry/runner/initializer.py

```diff
@@ -215,10 +221,8 @@ def apply_legacy_settings(settings):
 
     # Set ALLOWED_HOSTS if it's not already available
     if not settings.ALLOWED_HOSTS:
-        from urlparse import urlparse
-        urlbits = urlparse(settings.SENTRY_URL_PREFIX)
-        if urlbits.hostname:
-            settings.ALLOWED_HOSTS = (urlbits.hostname,)
+        from .hacks import AllowedHosts
+        settings.ALLOWED_HOSTS = AllowedHosts()
```

Comment:
Yeah, for record, I'm pretty sure ALLOWED_HOSTS is only used for url reversing from `HttpRequest.get_host()`.

I opted to patch it at this level first instead since it seemed less disruptive.